src/libsync/propagateuploadng.cpp

```diff
@@ -502,4 +502,16 @@ void PropagateUploadFileNG::slotUploadProgress(qint64 sent, qint64 total)
     }
     propagator()->reportProgress(*_item, _sent + sent - total);
 }
+
+
+void PropagateUploadFileNG::abort()
+{
+    foreach (AbstractNetworkJob *job, _jobs) {
+        // Abort only PUT and MKDIR jobs, since abording MoveJob
+        // might result in conflict.
+        if (job->reply() && !job->inherits("OCC::MoveJob")) {
```

Comment:
And to answer your question about stale directory, the next sync will resume the upload, so it won't be stale.